src/tally_filter.F90

```diff
@@ -1210,6 +1224,84 @@ function text_label_azimuthal(this, bin) result(label)
   end function text_label_azimuthal
 
 !===============================================================================
+! EnergyFunctionFilter methods
+!===============================================================================
+  subroutine get_next_bin_energyfunction(this, p, estimator, current_bin, &
+       next_bin, weight)
+    class(EnergyFunctionFilter), intent(in)  :: this
+    type(Particle),              intent(in)  :: p
+    integer,                     intent(in)  :: estimator
+    integer, value,              intent(in)  :: current_bin
+    integer,                     intent(out) :: next_bin
+    real(8),                     intent(out) :: weight
+
+    integer :: n, indx
+    real(8) :: E, f
+
+    select type(this)
+    type is (EnergyFunctionFilter)
+      if (current_bin == NO_BIN_FOUND) then
+        n = size(this % energy)
+
+        ! Make sure the correct energy is used.
+        if (estimator == ESTIMATOR_TRACKLENGTH) then
+          E = p % E
+        else
+          E = p % last_E
+        end if
+
+        ! Check if energy of the particle is within energy bins.
+        if (E < this % energy(1) .or. E > this % energy(n)) then
+          next_bin = NO_BIN_FOUND
+          weight = ONE
```

Comment:
Ah, ok, that makes sense now. Yes, perhaps `ERROR_REAL` is a little more obvious.